qi-lib/flow/core/deforest.rkt

```diff
@@ -180,17 +180,17 @@
   ;; producer. Procedures accepting variable number of arguments like
   ;; `map` cannot be in this class.
   (define-syntax-class fusable-stream-transformer0
-    #:attributes (f next)
     #:datum-literals (#%host-expression #%blanket-template __ _ #%fine-template)
     (pattern (~or (#%blanket-template
-                   ((#%host-expression (~datum filter))
+                   ((#%host-expression (~or (~datum filter)
+                                            (~datum filter-map)))
                     (#%host-expression f)
                     __))
                   (#%fine-template
-                   ((#%host-expression (~datum filter))
+                   ((#%host-expression (~or (~datum filter)
+                                            (~datum filter-map)))
                     (#%host-expression f)
-                    _)))
-             #:attr next #'filter-cstream-next))
+                    _)))))
 
   ;; All implemented stream transformers - within the stream, only
```

Comment:
This should really be an unspaced em-dash, but fixed-width fonts… my preference
would therefore be a triple-dash, as in "[…] stream transformers---within the
stream […]."